tools/generator/src/DTO/BazelLabel.swift

```diff
@@ -11,10 +11,20 @@ extension BazelLabel {
     static let nameSeparator = ":"
     static let packageSeparator = "/"
 
-    init?(_ value: String) {
+    enum ParseError: Error {
+        case missingOrTooManyRootSeparators
+        case missingNameAndPackage
+        case tooManyColons
+    }
+
+    private static func parse(
+        _ value: String
+    ) throws -> (repository: String, package: String, name: String) {
+        // swiftlint:disable:previous large_tuple
+
```

Comment:
I updated the parse logic to throw errors to provide better feedback. In addition, I moved the parse logic to its own function so that it is easier to implement the other initializers.